web/client/components/geostory/layouts/sections/Banner.jsx

```diff
@@ -0,0 +1,166 @@
+ */
+import React from "react";
+import {compose, withStateHandlers} from 'recompose';
+import SectionContents from '../../contents/SectionContents';
+import Background from './Background';
+import {backgroundPropWithHandler} from './enhancers/immersiveBackgroundManager';
+import ContainerDimensions from 'react-container-dimensions';
+import AddBar from '../../common/AddBar';
+import { SectionTypes, Modes, MediaTypes, SectionTemplates } from '../../../../utils/GeoStoryUtils';
+import titlePattern from './patterns/dots.png';
+import coverPattern from './patterns/grid.svg';
+import {get} from 'lodash';
+/**
+ * Paragraph Section Type.
+ * Paragraph is a page block that expands for all it's height
+ */
+export default compose(
+    backgroundPropWithHandler,
+    withStateHandlers({textEditorActive: false}))(({
+    id,
+    background = {},
+    contents = [],
+    mode,
+    contentId,
+    path,
+    sectionType = SectionTypes.BANNER,
+    cover,
+    viewWidth,
+    viewHeight,
+    inViewRef,
+    add = () => {},
+    updateSection = () => {},
+    remove = () => {},
+    updateBackground = () => {},
+    editMedia = () => {},
+    focusedContent,
+    expandableMedia = false,
+    storyTheme,
+    mediaViewer,
+    contentToolbar,
+    inView,
+    bubblingTextEditing = () => {}
+}) => {
+    const hideContent = get(focusedContent, "target.id") === contentId;
+    const visibility = hideContent ?  'hidden' : 'visible';
+    const expandableBackgroundClassName = expandableMedia && background && background.type === 'map' ? ' ms-expandable-background' : '';
+    const deletePath = path.replace('.background', '');
+
+    return (
+        <section
+            ref={inViewRef}
+            className={`ms-section ms-section-banner${expandableBackgroundClassName}`}
+            id={id}
+        >
+            <ContainerDimensions>
+                {({ height }) =>
+                // when section height is less then the view height
+                // background height need to be equal to section size
+                // this is important when working with z-index of section
+                // in case we increase the z-index of title the whole background is visible and overlap next section
+                    <Background
+                        { ...background }
+                        // selector used by sticky polyfill to detect scroll events
+                        scrollContainerSelector="#ms-sections-container"
+                        key={background.id}
+                        mode={mode}
+                        id={contentId}
+                        path={path}
+                        update={updateBackground}
+                        add={add}
+                        editMedia={editMedia}
+                        expandable={expandableMedia}
+                        updateSection={updateSection}
+                        cover={cover}
+                        remove={{removeFunc: remove, customPath: deletePath}}
```

Comment:
Remove prop is expected as a function inside ToolbarButtons so instead to change the component we could use the function directly here in the prop eg:

```
remove={() => remove(deletePath)}
```